### ISSUE TITLE
Errno::EIO: Input/output error with config.build(quiet: false) 

### ISSUE THREAD

**Issue by Silex** [ianchesal/packer-config#8]
Hello,

I think there's a logic error in https://github.com/ianchesal/packer-config/blob/master/lib/packer/runner.rb#L21

With the following code:

``` ruby
  def make_install_config(iso, scripts)
    name = iso_name(iso)
    iso_checksum = `md5sum #{iso}`.split.first
    ssh_host_port = 2222
    ssh_key_path = "ssh/support-no-pwd"
    output = "output/install/%s" % name

    config = Packer::Config.new("install-%s.json" % name)

    builder = config.add_builder(Packer::Builder::VIRTUALBOX_ISO)
    builder.vm_name name
    builder.guest_os_type "Ubuntu_64"
    builder.guest_additions_mode "disable"
    builder.headless true
    builder.disk_size 20000
    builder.vboxmanage [
      [ "modifyvm", "{{.Name}}", "--memory", "2048" ],
      [ "modifyvm", "{{.Name}}", "--natnet1", "10.13.37/24" ],
      [ "modifyvm", "{{.Name}}", "--natpf1", "delete", "packerssh" ],
      [ "modifyvm", "{{.Name}}", "--natpf1", "packerssh,tcp,127.0.0.1,2222,10.13.37.4,22" ]
    ]
    builder.iso_url iso
    builder.iso_checksum iso_checksum
    builder.iso_checksum_type "md5"
    builder.boot_command ["I<enter>"]
    builder.boot_wait "5s"
    builder.ssh_username "root"
    builder.ssh_key_path ssh_key_path
    builder.ssh_host_port_min ssh_host_port
    builder.ssh_host_port_max ssh_host_port
    builder.ssh_wait_timeout "1200s"
    builder.shutdown_command "shutdown -h now"
    builder.output_directory output
    builder.format "ova"

    provisioner = config.add_provisioner Packer::Provisioner::SHELL
    provisioner.scripts scripts

    config
  end

  def test_install
    config = make_install_config(ISO_FILES.last, ["scripts/check_services.sh"])
    assert(config.validate)
    assert(config.build)
  end
```

Makes the following error:

```
philippe@philippe-desktop:~/work/stvs/provision$ ruby test.rb
Loaded suite test
Started
virtualbox-iso output will be in this color.

==> virtualbox-iso: Downloading or copying ISO
    virtualbox-iso: Downloading or copying: file:///iso/new/provision-5.8.4-2499-ac54cb2.iso
==> virtualbox-iso: Creating virtual machine...
==> virtualbox-iso: Creating hard drive...
==> virtualbox-iso: Creating forwarded port mapping for SSH (host port 2222)
==> virtualbox-iso: Executing custom VBoxManage commands...
    virtualbox-iso: Executing: modifyvm provision-5.8.4-2499-ac54cb2 --memory 2048
    virtualbox-iso: Executing: modifyvm provision-5.8.4-2499-ac54cb2 --natnet1 10.13.37/24
    virtualbox-iso: Executing: modifyvm provision-5.8.4-2499-ac54cb2 --natpf1 delete packerssh
    virtualbox-iso: Executing: modifyvm provision-5.8.4-2499-ac54cb2 --natpf1 packerssh,tcp,127.0.0.1,2222,10.13.37.4,22
==> virtualbox-iso: Starting the virtual machine...
    virtualbox-iso: WARNING: The VM will be started in headless mode, as configured.
    virtualbox-iso: In headless mode, errors during the boot sequence or OS setup
    virtualbox-iso: won't be easily visible. Use at your own discretion.
==> virtualbox-iso: Waiting 5s for boot...
==> virtualbox-iso: Typing the boot command...
==> virtualbox-iso: Waiting for SSH to become available...
==> virtualbox-iso: Connected to SSH!
==> virtualbox-iso: Uploading VirtualBox version info (4.3.10)
==> virtualbox-iso: Provisioning with shell script: scripts/check_services.sh
    virtualbox-iso: Checking service mysql
    virtualbox-iso: mysql start/running, process 1017
    virtualbox-iso: Checking service provision
    virtualbox-iso: provision start/running, process 1232
    virtualbox-iso: Checking service rstreamer
    virtualbox-iso: rstreamer start/running, process 367
    virtualbox-iso: Trying a MySQL query
    virtualbox-iso: count(*)
    virtualbox-iso: 2
==> virtualbox-iso: Gracefully halting virtual machine...
==> virtualbox-iso: Preparing to export machine...
    virtualbox-iso: Deleting forwarded port mapping for SSH (host port 2222)
==> virtualbox-iso: Exporting virtual machine...
    virtualbox-iso: Executing: export provision-5.8.4-2499-ac54cb2 --output output/install/provision-5.8.4-2499-ac54cb2/provision-5.8.4-2499-ac54cb2.ova
==> virtualbox-iso: Unregistering and deleting virtual machine...
Build 'virtualbox-iso' finished.

==> Builds finished. The artifacts of successful builds are:
--> virtualbox-iso: VM files in directory: output/install/provision-5.8.4-2499-ac54cb2
E
===============================================================================
Error: test_install(ProvisionTests)
: Errno::EIO: Input/output error @ io_fillbuf - fd:7 /dev/pts/0
/usr/local/bundle/gems/packer-config-1.3.0/lib/packer/runner.rb:21:in `eof?'
/usr/local/bundle/gems/packer-config-1.3.0/lib/packer/runner.rb:21:in `block in run!'
/usr/local/bundle/gems/packer-config-1.3.0/lib/packer/runner.rb:40:in `spawn'
/usr/local/bundle/gems/packer-config-1.3.0/lib/packer/runner.rb:40:in `pty'
/usr/local/bundle/gems/packer-config-1.3.0/lib/packer/runner.rb:20:in `run!'
/usr/local/bundle/gems/packer-config-1.3.0/lib/packer-config.rb:107:in `block in build'
/usr/local/bundle/gems/packer-config-1.3.0/lib/packer-config.rb:105:in `chdir'
/usr/local/bundle/gems/packer-config-1.3.0/lib/packer-config.rb:105:in `build'
```

In the example at http://ruby-doc.org/stdlib-2.1.0/libdoc/pty/rdoc/PTY.html it mentions that rescue GNU/Linux raises Errno::EIO when io dies.

Using `config.build(quiet: true)` makes it work but it's not very desirable.

**Comment by Silex** [comment]
Based on http://stackoverflow.com/questions/1154846/continuously-read-from-stdout-of-external-process-in-ruby I'm trying to add a silent `rescue Errno::EIO` to see if it works.

**Comment by ianchesal** [comment]
Hmm. We can probably lost the pseudo-terminal altogether there. I see your PR to patch it but let me take a swing at just not having a PTY involved at all.

**Comment by Silex** [comment]
No worries, I'll close this and wait in #9

**Comment by ianchesal** [comment]
@Silex I just pushed a change to the `develop` branch -- want to pull the code and try it? It removes ptys from the runner.lib altogether.

**Comment by Silex** [comment]
Yeah it looks alright, there's a typo in the commit message body tho ("Psuedo").

**Comment by ianchesal** [comment]
Released 1.3.1 to fix this.
